pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -482,6 +482,12 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 		if err != nil {
 			return nil, err
 		}
+		defer func() {
+			if err := gce.deleteStaticIP(name, region); err != nil {
```

Comment:
yeah, if the delete actually fails, we will still leak, but this seems strictly better than what we have today, no?